File: articles/systems-modeling/python/assumption_risk.py
"""
Systems Modeling:
Assumption risk scoring.

Educational example only.
"""

from __future__ import annotations

import pandas as pd


def main() -> None:
    assumptions = pd.read_csv("../data/model_assumptions.csv")

    assumptions["assumption_risk_score"] = (
        (1.0 - assumptions["confidence"]) * assumptions["impact_if_wrong"]
    )

    assumptions = assumptions.sort_values("assumption_risk_score", ascending=False)

    print(assumptions)

    assumptions.to_csv("../outputs/python_systems_modeling_assumption_risk.csv", index=False)


if __name__ == "__main__":
    main()

File: articles/systems-modeling/r/assumption_risk.R
# Systems Modeling:
# Assumption risk scoring in R.
# Educational example only.

library(tidyverse)

assumptions <- read_csv("../data/model_assumptions.csv", show_col_types = FALSE)

scored <- assumptions |>
  mutate(
    assumption_risk_score = (1 - confidence) * impact_if_wrong,
    priority = case_when(
      assumption_risk_score >= 0.30 ~ "High",
      assumption_risk_score >= 0.18 ~ "Medium",
      TRUE ~ "Lower"
    )
  ) |>
  arrange(desc(assumption_risk_score))

dir.create("../outputs", showWarnings = FALSE, recursive = TRUE)

write_csv(scored, "../outputs/r_systems_modeling_assumption_risk.csv")

print(scored)
